aten/src/ATen/SparseTensorImpl.h

```diff
@@ -80,8 +99,15 @@ struct SparseTensorImpl : public TensorImpl {
     }
     sparseDims_ = sparseDims;
     denseDims_ = denseDims;
+#else
+    AT_CHECK(sparseDims + denseDims == size.size(), "number of dimensions must be sparseDims (", sparseDims, ") + denseDims (", denseDims, "), but got ", size.size());
```

Comment:
I did a bit more refactoring and move the `resize_and_clear_` function into SparseTensorImpl.h to replace the `raw_resize_` function. In the `resize_and_clear_` function, since we are going to assign empty indices and values to the resized tensor, we won't need to check their original shapes.